sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -454,6 +454,67 @@ case class StringReplace(srcExpr: Expression, searchExpr: Expression, replaceExp
   override def prettyName: String = "replace"
 }
 
+// scalastyle:off line.size.limit
+@ExpressionDescription(
+  usage = "_FUNC_(input, replace, pos[, len]) - Replace `input` with `replace` that starts at `pos` and is of length `len`.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_('Spark SQL' PLACING '_' FROM 6);
+       Spark_SQL
+      > SELECT _FUNC_('Spark SQL' PLACING 'CORE' FROM 7);
+       Spark CORE
+      > SELECT _FUNC_('Spark SQL' PLACING 'ANSI ' FROM 7 FOR 0);
+       Spark ANSI SQL
+      > SELECT _FUNC_('Spark SQL' PLACING 'tructured' FROM 2 FOR 4);
+       Structured SQL
+  """)
+// scalastyle:on line.size.limit
+case class Overlay(input: Expression, replace: Expression, pos: Expression, len: Expression)
+  extends QuaternaryExpression with ImplicitCastInputTypes with NullIntolerant {
+
+  def this(str: Expression, replace: Expression, pos: Expression) = {
+    this(str, replace, pos, Literal(Integer.MAX_VALUE))
+  }
+
+  override def dataType: DataType = StringType
+
+  override def inputTypes: Seq[AbstractDataType] =
+    Seq(StringType, StringType, IntegerType, IntegerType)
+
+  override def children: Seq[Expression] = input :: replace :: pos :: len :: Nil
+
+  override def nullSafeEval(inputEval: Any, replaceEval: Any, posEval: Any, lenEval: Any): Any = {
+    val inputStr = inputEval.asInstanceOf[UTF8String]
+    val replaceStr = replaceEval.asInstanceOf[UTF8String].toString
+    val position = posEval.asInstanceOf[Int]
+    var length = lenEval.asInstanceOf[Int]
+    if (length.equals(Int.MaxValue)) {
+      length = replaceStr.size
+    }
+    val headStr = inputStr.substringSQL(1, position - 1)
+    val tailStr = inputStr.substringSQL(position + length, Int.MaxValue)
+    UTF8String.fromString(headStr.toString + replaceStr + tailStr.toString)
+  }
+
+  override def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
+    val length = ctx.addMutableState(CodeGenerator.JAVA_INT, "length")
+    val head = ctx.addMutableState("UTF8String", "head")
+    val tail = ctx.addMutableState("UTF8String", "tail")
+    nullSafeCodeGen(ctx, ev, (input, place, pos, len) => {
+      val isMax = s"$len.equals(${Int.MaxValue})"
+      s"""if ($isMax) {
+        $length = $replace.toString().size;
+      } else {
+        $length = $len;
+      }
+      $head = $input.substringSQL(1, $pos - 1);
+      $tail = $input.substringSQL($pos + $length, ${Int.MaxValue});
+      ${ev.value} = $head + $replace + $tail;
+      """
```

Comment:
Btw, could you follow the string format for codes like:

```
s"""
   |...
   |...
 """.stripMargin
```

?

Reply:
@ueshin Thanks for your reminder. I changed to the format.